src/test/app/ValidatorSite_test.cpp

```diff
@@ -97,6 +104,11 @@ class http_sync_server
     {
         error_code ec;
         acceptor_.close(ec);
```

Comment:
@wilsonianb Calling `close` will cause the `on_accept` callback to be called. Before this patch, that would access the member variable `accessor_`, which may happen after the destructor has completed running. So there's definitely an issue in the existing code.

However, I just ran it and `ec` is always set in the callback when `close` is called. I should be able to get rid of the condition variable as long as I check the error code first. That will simplify this code. I'll submit a new patch shortly.